engine/src/main/java/org/terasology/rendering/dag/nodes/DirectionalLightsNode.java

```diff
@@ -152,7 +153,7 @@ private void applyLightBufferPass() {
         bindDisplay();     // TODO: verify this is necessary
         setViewportToSizeOf(sceneOpaque);    // TODO: verify this is necessary
 
-        frameBuffersManager.swapSceneOpaqueFBOs();
+        dynamicFBM.swap(DefaultDynamicFBOs.WriteOnlyGBuffer.getResourceUrn(), DefaultDynamicFBOs.ReadOnlyGBuffer.getResourceUrn());
```

Comment:
I think the method should certainly send notifications. If you think about the BindFBO state changes, they need to know if the fboIDs behind read/writeOnlyGBuffer has changed so that they can do the change in the associated task.

Regarding the pairing, I guess you could hardcode the swapping to happen only between these two default FBOs for the time being. But it'd be good if the FBMs had a method to create paired FBOs: ultimately they'd have the same FBOconfig, just different fboNames. Perhaps it wouldn't be too difficult.